File: backend/app/schemas/sleep.py
from pydantic import BaseModel, ConfigDict
from datetime import date, datetime
from typing import List, Dict, Any

class SleepListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    calendar_date: date
    sleep_time_seconds: int | None
    sleep_score: int | None
    sleep_score_feedback: str | None
    average_hrv: float | None
    average_resting_heart_rate: int | None
    sleep_heart_rate: List[Dict[str, Any]] | None = None
    sleep_stress: List[Dict[str, Any]] | None = None

class SleepListResponse(BaseModel):
    sleep_records: List[SleepListItem]
    total: int
    page: int
    page_size: int
    has_more: bool

class SleepDetailResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    calendar_date: date
    sleep_time_seconds: int | None
    nap_time_seconds: int | None
    awake_time_seconds: int | None
    deep_sleep_seconds: int | None
    light_sleep_seconds: int | None
    rem_sleep_seconds: int | None
    unmeasurable_sleep_seconds: int | None
    sleep_start: datetime | None
    sleep_end: datetime | None
    sleep_score: int | None
    sleep_score_feedback: str | None
    sleep_score_insight: str | None
    average_hrv: float | None
    average_resting_heart_rate: int | None
    average_sp_o2: float | None
    raw_data: Dict[str, Any] | None

File: frontend/types/sleep.ts
export interface SleepListItem {
  id: number;
  calendar_date: string;
  sleep_time_seconds: number | null;
  sleep_score: number | null;
  sleep_score_feedback: string | null;
  average_hrv: number | null;
  average_resting_heart_rate: number | null;
  sleep_heart_rate: { value: number; startGMT: number }[] | null;
  sleep_stress: { value: number; startGMT: number }[] | null;
}

export interface SleepListResponse {
  sleep_records: SleepListItem[];
  total: number;
  page: number;
  page_size: number;
  has_more: boolean;
}

export interface SleepDetail {
  id: number;
  calendar_date: string;
  sleep_time_seconds: number | null;
  nap_time_seconds: number | null;
  awake_time_seconds: number | null;
  deep_sleep_seconds: number | null;
  light_sleep_seconds: number | null;
  rem_sleep_seconds: number | null;
  unmeasurable_sleep_seconds: number | null;
  sleep_start: string | null;
  sleep_end: string | null;
  sleep_score: number | null;
  sleep_score_feedback: string | null;
  sleep_score_insight: string | null;
  average_hrv: number | null;
  average_resting_heart_rate: number | null;
  average_sp_o2: number | null;
  raw_data: any;
}
